tensorflow/cc/training/gradient_descent_optimizer.cc

```diff
@@ -0,0 +1,33 @@
+/* Copyright 2015 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/cc/training/gradient_descent_optimizer.h"
+
+namespace tensorflow {
+
+Output GradientDescentOptimizer::ApplyDense(const Scope& scope,
+                                            const Output& grad,
+                                            const Output& var) const {
+    // TODO(theflofly): better way to convert a data type ref
+    // to a data type then static_cast<DataType>(var.type() - 100)) ?
+    return tensorflow::ops::ApplyGradientDescent(scope.NewSubScope("update"),
+                                                 {var},
+                                                 tensorflow::ops::Cast(scope.NewSubScope("learning_rate"),
+                                                                       learning_rate_,
```

Comment:
Please fix indentation and conform to Google C++ style.
https://google.github.io/styleguide/cppguide.html

You can also find instructions on how to run clang-tidy on the code here:
https://github.com/tensorflow/tensorflow/blob/master/CONTRIBUTING.md#c-coding-style